api/auth/auth.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * @license
+ * Copyright 2025 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auth
+
+import (
+	"context"
+	"net/http"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+func NewTokenBasedClient(ctx context.Context, apiToken string) *http.Client {
+	return oauth2.NewClient(ctx, oauth2.StaticTokenSource(&oauth2.Token{TokenType: "Api-Token", AccessToken: apiToken}))
+}
+
+func NewPlatformTokenClient(ctx context.Context, platformToken string) *http.Client {
```

Comment:
Nit: either we should add "Based" to this function name or remove it from the other two.